### PR TITLE
fix(k8sprocessor): retrieve deployment name from API server

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: pkg/processor/k8sprocessor/kube/client.go [hunk 1]
```diff
@@ -294,8 +294,10 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 				if c.Rules.DaemonSetName {
 					tags[c.Rules.Tags.DaemonSetName] = owner.name
 				}
-			case "DeploymentName":
-				// This should be already set earlier
+			case "Deployment":
```
Comment (pmm-sumo) [inline pkg/processor/k8sprocessor/kube/client.go:297]: So the name of the attribute changed?

**Review by sumo-drosiek — APPROVED** [review]
LGTM

File: pkg/processor/k8sprocessor/kube/client_test.go [hunk 2]
```diff
@@ -586,16 +602,22 @@ func TestExtractionRules(t *testing.T) {
...
 			p, ok := c.GetPod(PodIdentifier(pod.Status.PodIP))
 			require.True(t, ok)
 
 			assert.Equal(t, len(tc.attributes), len(p.Attributes))
 			for k, v := range tc.attributes {
 				got, ok := p.Attributes[k]
-				assert.True(t, ok)
-				assert.Equal(t, v, got)
+				assert.True(t, ok, "Attribute '%s' not found.", k)
+				assert.Equal(t, v, got, "Value of '%s' should be '%s', but was '%s'.", k, v, got)
```
Comment (pmalek-sumo) [inline pkg/processor/k8sprocessor/kube/client_test.go:620]: These additional log should be included in the message from `testify`, aren't they?

File: pkg/processor/k8sprocessor/kube/client.go (hunk 1 above)
Comment (astencel-sumo) [reply pkg/processor/k8sprocessor/kube/client.go:282]: No, this line was incorrect already before :) Note that this switch statement operates not on attribute names, but on Kubernetes resource kind names (Deployment, ReplicaSet, StatefulSet, etc.).

File: pkg/processor/k8sprocessor/kube/client_test.go (hunk 2 above)
Comment (astencel-sumo) [reply pkg/processor/k8sprocessor/kube/client_test.go:632]: Yes, these messages are displayed when the assertions fail. The attribute name is crucial here - without these additional messages, it's not possible to see which attribute the assertion failed for.

Example output:

```console
$ go test
--- FAIL: TestExtractionRules (0.00s)
    --- FAIL: TestExtractionRules/deployment_name_with_owner_lookup (0.00s)
        client_test.go:632: 
                Error Trace:    client_test.go:632
                Error:          Not equal: 
                                expected: "snug-sts"
                                actual  : "dearest-deploy"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -snug-sts
                                +dearest-deploy
                Test:           TestExtractionRules/deployment_name_with_owner_lookup
                Messages:       Value of 'k8s.deployment.name' should be 'snug-sts', but was 'dearest-deploy'.
FAIL
exit status 1
FAIL    github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/kube   0.610s
```

**Review by pmalek-sumo — APPROVED** [review]

**Review by kkujawa-sumo — APPROVED** [review]

**Pull request merged**